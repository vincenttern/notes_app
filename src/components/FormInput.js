import React, { Component } from "react";
const uniqid = require("uniqid");

class FormInput extends Component {
  state = {
    textInput: "",
    dbStorage: []
  };

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleFormSubmit = event => {
    event.preventDefault();

    let newObj = {};
    let newArr = [];

    newArr.push(uniqid(), this.state.textInput);
    newObj[newArr[0]] = newArr[1];

    this.setState(
      {
        dbStorage: this.state.dbStorage.concat(newObj)
      },
      () => {
        localStorage.setItem("Notes", JSON.stringify(this.state.dbStorage));
        console.log(this.state.dbStorage);
      }
    );
  };

  componentDidMount() {
    const retreiveSavedItems = localStorage.getItem("Notes");
    const parseSavedItems = JSON.parse(retreiveSavedItems);

    if (parseSavedItems) {
      this.setState({
        dbStorage: parseSavedItems
      });
    }
  }

  render() {
    return (
      <div>
        <main>
          <form onSubmit={this.handleFormSubmit}>
            <label>
              <input
                type="text"
                name="textInput"
                value={this.state.textInput}
                onChange={this.handleChange}
                placeholder="write something..."
              />
            </label>
            <input type="submit" value="Submit" />
          </form>
        </main>

        <div>
          <p>{this.state.textInput}</p>
        </div>
      </div>
    );
  }
}

export default FormInput;
