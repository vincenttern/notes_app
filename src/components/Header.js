import React from "react";
import headerStyles from "../styles/scss/Header.module.scss";

const Header = () => {
  return (
    <header className={headerStyles.header}>
      <h2 className={headerStyles.title}>NoteBeast</h2>
    </header>
  );
};

export default Header;
