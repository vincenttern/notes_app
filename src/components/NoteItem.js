import React, { Component } from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import Breadcrumb from "react-bootstrap/Breadcrumb";

import EditIcon from "@material-ui/icons/Edit";
import DeleteForeverIcon from "@material-ui/icons/DeleteForever";

import { confirmAlert } from "react-confirm-alert"; // Import
import "react-confirm-alert/src/react-confirm-alert.css"; // Import css

import noteItemStyles from "../styles/scss/NoteItem.module.scss";

export default class NoteItem extends Component {
  render() {
    const { data, date, title, handleDelete, handleEdit } = this.props;

    let editedItemTitle;
    if (data.filteredEditItem) {
      editedItemTitle = data.myEditItem;
    }

    return (
      <li className={noteItemStyles.li}>
        <Card style={{ width: "100%" }}>
          <Card.Body>
            <Card.Text>
              {data.changedEditTitle ? editedItemTitle : title}
            </Card.Text>
            <Breadcrumb>
              <Breadcrumb.Item>
                <Button
                  variant="outline-warning"
                  size="sm"
                  onClick={handleEdit}
                >
                  <EditIcon />
                </Button>
              </Breadcrumb.Item>
              <Breadcrumb.Item>
                <Button
                  variant="outline-danger"
                  size="sm"
                  onClick={() =>
                    confirmAlert({
                      title: "Confirm to submit",
                      message: "Are you sure to do this.",
                      buttons: [
                        {
                          label: "Yes",
                          onClick: () => handleDelete("Click OK To Delete")
                        },
                        {
                          label: "No"
                        }
                      ]
                    })
                  }
                >
                  <DeleteForeverIcon />
                </Button>
              </Breadcrumb.Item>
              <Breadcrumb.Item active>
                <Button variant="outline-info" disabled>
                  {date}
                </Button>
              </Breadcrumb.Item>
            </Breadcrumb>
          </Card.Body>
        </Card>
      </li>
    );
  }
}
