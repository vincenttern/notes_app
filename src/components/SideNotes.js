import React, { Component } from "react";
import ListGroup from "react-bootstrap/ListGroup";

import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

export default class SideNotes extends Component {
  render() {
    const { data, sideItemClicked } = this.props;
    var truncateTitles = [];

    data.dbStorage.forEach(item => {
      if (item.title.length > 1 && item.date) {
        let truncateTitle = item.title.substring(0, 40) + "...";
        let slicedDate = item.date.slice(5, 9);
        truncateTitles.push({
          title: truncateTitle,
          cutDate: slicedDate,
          id: item.id
        });
      }
    });
    return (
      <ListGroup variant="flush">
        <h3>My Notes</h3>
        {truncateTitles.map(item => {
          return (
            <ListGroup.Item
              key={item.id}
              action
              variant="light"
              onClick={() => sideItemClicked(item.id)}
            >
              <Container>
                <Row>
                  <Col sm={11}>{item.title}</Col>
                  <Col sm={1}>{item.cutDate}</Col>
                </Row>
              </Container>
            </ListGroup.Item>
          );
        })}
      </ListGroup>
    );
  }
}
