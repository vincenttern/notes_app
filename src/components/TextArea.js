import React, { Component } from "react";
import Button from "react-bootstrap/Button";

export default class TextArea extends Component {
  render() {
    const { data, handleChange, handleSubmit, alertUser } = this.props;

    return (
      <form onSubmit={!data.textInput ? alertUser : handleSubmit}>
        <textarea
          label="Note"
          placeholder="Take down some notes....."
          value={data.textInput}
          onChange={handleChange}
        />

        <Button
          variant={data.editItem ? "warning" : "primary"}
          size="lg"
          block
          type="submit"
        >
          {data.editItem ? "Edit Note" : "Add Note"}
        </Button>
      </form>
    );
  }
}
