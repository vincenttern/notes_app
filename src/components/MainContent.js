import React from "react";
import FormInput from "../components/FormInput";
// import TextField from "@material-ui/core/TextField";

function MainContent() {
  return (
    <div>
      <main>
        <FormInput />
      </main>
    </div>
  );
}

export default MainContent;
