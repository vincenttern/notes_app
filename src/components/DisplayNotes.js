import React, { Component } from "react";
import NoteItem from "./NoteItem";

export default class DisplayNotes extends Component {
  render() {
    const { data, handleDelete, handleEdit } = this.props;

    const displayNotes = {
      width: "50%",
      position: "relative",
      left: "50%",
      transform: "translate(-50%, 0)"
    };

    const yourNotes = {
      fontSize: 30,
      textAlign: "center"
    };

    return (
      <ul style={displayNotes}>
        <h3 style={yourNotes}> My Notes</h3>
        {data.dbStorage.map(item => {
          return (
            <NoteItem
              key={item.id}
              title={item.title}
              handleEdit={() => handleEdit(item.id)}
              handleDelete={() => handleDelete(item.id)}
            />
          );
        })}
      </ul>
    );
  }
}
