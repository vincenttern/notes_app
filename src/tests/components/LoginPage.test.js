import React from "react";
import { shallow } from "enzyme";
import { LoginPage } from "../../components/LoginPage";

test("should render LoginPage correctly", () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});

test("should call startLoginWithGoogle on button click", () => {
  const startLoginWithGoogle = jest.fn();
  const wrapper = shallow(
    <LoginPage startLoginWithGoogle={startLoginWithGoogle} />
  );
  wrapper
    .find("button")
    .at(0)
    .simulate("click");
  expect(startLoginWithGoogle).toHaveBeenCalled();
});

test("should call startLoginWithFacebook on button click", () => {
  const startLoginWithFacebook = jest.fn();
  const wrapper = shallow(
    <LoginPage startLoginWithFacebook={startLoginWithFacebook} />
  );
  wrapper
    .find("button")
    .at(1)
    .simulate("click");
  expect(startLoginWithFacebook).toHaveBeenCalled();
});
