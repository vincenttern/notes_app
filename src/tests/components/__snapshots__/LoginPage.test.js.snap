// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      React Boiler Plate
    </h1>
    <p>
      Tag line for app
    </p>
    <div
      className="button-box__logins"
    >
      <button
        className="button"
      >
        Login with Google
      </button>
      <button
        className="button"
      >
        Login with Facebook
      </button>
    </div>
  </div>
</div>
`;
